Frameworks/AddressBook/ABRecord.mm

```diff
@@ -80,18 +152,404 @@ ABRecordType ABRecordGetRecordType(ABRecordRef record) {
+            NSDate* value = (__bridge_transfer NSDate*)[multiValue copyValueAtIndex:i];
+            WACContactDate* date = [WACContactDate make];
+            date.kind = dict[label] == nil ? WACContactDateKindOther : [dict[label] integerValue];
+
+            unsigned int units = NSYearCalendarUnit | NSMonthCalendarUnit | NSDayCalendarUnit;
+            NSCalendar* calendar = [NSCalendar currentCalendar];
+            NSDateComponents* components = [calendar components:units fromDate:value];
+
+            date.year = @([components year]);
+            date.month = @([components month]);
+            date.day = @([components day]);
+
+            [importantDates addObject:date];
+        }
+    } else if (contactProperty == kABPersonPhoneProperty) {
+        _ABMultiValue* multiValue = (__bridge _ABMultiValue*)value;
+        if (!_checkType(kABStringPropertyType, multiValue, error)) {
+            return false;
+        }
+
+        // Ensure that none of the values violate the allowed length of Windows
+        // Contacts properties.
+        for (int i = 0; i < [multiValue getCount]; i++) {
+            NSString* value = (__bridge_transfer NSString*)[multiValue copyValueAtIndex:i];
+            if (!_checkLength(value, kABPersonPhoneLength, error, @"phone")) {
+                return false;
+            }
+        }
+
+        NSMutableArray* phones = person.contact.phones;
+        [phones removeAllObjects];
+
+        NSDictionary* dict = @{
+            ((__bridge NSString*)kABHomeLabel) : @(WACContactPhoneKindHome),
+            ((__bridge NSString*)kABPersonPhoneMobileLabel) : @(WACContactPhoneKindMobile),
+            ((__bridge NSString*)kABWorkLabel) : @(WACContactPhoneKindWork),
+            ((__bridge NSString*)kABPersonPhonePagerLabel) : @(WACContactPhoneKindPager),
+            ((__bridge NSString*)kABPersonPhoneWorkFAXLabel) : @(WACContactPhoneKindBusinessFax),
+            ((__bridge NSString*)kABPersonPhoneHomeFAXLabel) : @(WACContactPhoneKindHomeFax),
+            ((__bridge NSString*)kABPersonPhoneCompanyLabel) : @(WACContactPhoneKindCompany),
+            ((__bridge NSString*)kABPersonPhoneAssistantLabel) : @(WACContactPhoneKindAssistant),
+            ((__bridge NSString*)kABPersonPhoneRadioLabel) : @(WACContactPhoneKindRadio),
+            ((__bridge NSString*)kABOtherLabel) : @(WACContactPhoneKindOther)
+        };
+
+        for (int i = 0; i < [multiValue getCount]; i++) {
+            NSString* label = (__bridge_transfer NSString*)[multiValue copyLabelAtIndex:i];
+            NSString* value = (__bridge_transfer NSString*)[multiValue copyValueAtIndex:i];
+            WACContactPhone* phone = [WACContactPhone make];
+            phone.number = value;
+            phone.kind = dict[label] == nil ? WACContactPhoneKindOther : [dict[label] integerValue];
+            [phones addObject:phone];
+        }
+    } else if (contactProperty == kABPersonURLProperty) {
```

Comment:
The documentation doesn't list a length limit for the String of a URL.

---
In reply to: [77101644](https://github.com/Microsoft/WinObjC/pull/935#discussion_r77101644) [](ancestors = 77101644)